lib/handlers/targetHandler.js

```diff
@@ -40,18 +40,31 @@ const isMatchingRegex = function (target, targetRegex) {
   );
 };
 
+const prependHttp = function(targetUrl){
+	if (targetUrl && url.parse(targetUrl).host === null)
+	{
+		targetUrl = 'http://' + targetUrl;
+	}
+	return targetUrl;
+};
+
 const isMatchingUrl = function (target, targetUrl) {
   if (targetUrl instanceof RegExp) {
     return false;
   }
+
+  targetUrl = prependHttp(targetUrl);
```

Comment:
`prependHttp` is parsing the url again, while the url is getting parsed in line 58 in the calling method.

I wonder if line 61 can simply return `http` as the default host?

Reply:
```
const url = require('url');
console.log(url.parse('amazon.com'));
console.log(url.parse('http://amazon.com'));
```

> Url {
  protocol: null,
  slashes: null,
  auth: null,
  host: null,
  port: null,
  hostname: null,
  hash: null,
  search: null,
  query: null,
  pathname: 'amazon.com',
  path: 'amazon.com',
  href: 'amazon.com'
}
Url {
  protocol: 'http:',
  slashes: true,
  auth: null,
  host: 'amazon.com',
  port: null,
  hostname: 'amazon.com',
  hash: null,
  search: null,
  query: null,
  pathname: '/',
  path: '/',
  href: 'http://amazon.com/'
}

As we observe, parsing without protocol results in different output.

I see the point of operation being done twice but resultant is different with(or out) protocol, If performance is the main concern we can parse the targetUrl before passing it to isMatchingUrl().

Would that be way to go?

Reply:
wasn't thinking of performance, it's a simple operation when seen in context of the usecase.

I was thinking of the code duplication.

Reply:
If Code duplication is of concern, not sure how do we fix that!

It looks hacky to me, may be we can do normal string comparison if `targetUrl` has prefix of 'http'? It is another workaround to surpass `url.parse()`

Reply:
I get it. I think in that case we can live with this. String pattern match is worse.

Reply:
I get it. I think in that case we can live with this. String pattern match is worse.